app/Services/Api/ApiPermissionService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+
+namespace App\Services\Api;
+
+use App\Models\CampaignPermission;
+use App\Models\CampaignRole;
+use App\Models\Entity;
+
+use Illuminate\Support\Facades\DB;
+
+class ApiPermissionService
+{
+    /**
+     * Set the entity type
+     * @param string $type
+     * @return PermissionService
+     */
+    public function type(int $type): self
+    {
+        $this->type = $type;
+        return $this;
+    }
+
+    /**
+     * Set the role
+     * @param CampaignRole $role
+     * @return $this
+     */
+    public function role(CampaignRole $role): self
+    {
+        $this->role = $role;
+        return $this;
+    }
+
+     /**
+     * Get the permissions of an entity
+     * @param Entity $entity
+     * @return mixed
+     */
+    public function entityPermissions(Entity $entity): array
```

Comment:
Since this is only called in the service, it's worth making the function `protected` instead of `public`.